### DISCUSSION TITLE
GA / version 1.0

### DISCUSSION THREAD

**Discussion by hermanbanken** [open-telemetry/opentelemetry-collector-contrib#2664] [General]
Where can I find how mature this collector is? We want to use this in production, but the v0.22 is a bit offsetting.

Is there any indication when version 1 will be released? There seem to be no open issues that are assigned to it:
https://github.com/open-telemetry/opentelemetry-collector-contrib/milestone/3

**Comment by hermanbanken** [comment]
Should we look at https://github.com/open-telemetry/opentelemetry-collector/milestones as releases are paired?

**Comment by mx-psi** [comment]
I believe the most up to date information can be found in this document https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/ga-roadmap.md The current focus is on bringing traces to GA, other signals (metrics and logs) will take more time (since the spec and protocol are not finalized yet)

**Reply by hermanbanken** [reply]
Thanks!

**Comment by YohanSciubukgian** [comment]
Do you have any update regarding the collector GA release timeline ?
What are the remaining projects before the collector could be considered GA ?

**Reply by mx-psi** [reply]
We are first working on the opentelemetry-collector modules on a module by module basis. For the first two modules under consideration you can check https://github.com/open-telemetry/opentelemetry-collector/issues/8978 and https://github.com/open-telemetry/opentelemetry-collector/issues/8946 for progress. 

We may update the GA roadmap document for a more up to date roadmap.

**Reply by YohanSciubukgian** [reply]
Thanks for the feedback. Could you please let us know when the roadmap update will be provided?

**Reply by mx-psi** [reply]
> Thanks for the feedback. Could you please let us know when the roadmap update will be provided?

I don't have a timeline for that, sorry. I personally intend to work on this, but I want first to understand our experience with these two modules which may shape our roadmap. In any case, I would be interested in knowing how the roadmap would be useful for you: 
- do you intend to use the Collector as a library (e.g. for building your own distro of the Collector or building custom Collector components)?
- do you instead intend to use the Collector as an end-users with a pre-built distro (or one that does not need custom components) and are interested in stability regarding its features?

**Reply by YohanSciubukgian** [reply]
I will be in the last scenario.
I'm looking for stability as I have a scenario with an embedded software that could not be often updated.
As the current collector releases are done every 2 weeks, with many of them including breaking changes, the ideal situation for me would be to have my embedded software using the latest JAVA library that aligns with the Collector v1 GA release so I could ensure that it will guarantee the compatibility over a longer period of time.
As interoperability between systems and languages are keys of what we are doing, embracing the OpenTelemetry project is making sense for the future of our projects.

**Reply by mx-psi** [reply]
Alright, thanks! I think the current roadmap is not very helpful to you then, since it is mostly focused on the Go API side of things. Will take this into consideration when updating the roadmap and try and keep you posted :)
